### PR TITLE
RubyCompatCheck: new check for new USE_RUBY compatible values

### PR DESCRIPTION
**Pull request by arthurzam**
I think the `ruby-ng.eclass` is a real abomination I've created. I've extracted parts from real ruby-ng, modified as I needed, took shortcuts and more. I hope no sane person will use this abomination, we need to cast in into the fire :)

Resolves: https://github.com/pkgcore/pkgcheck/issues/304

### CODE REVIEW

File: src/pkgcheck/checks/ruby.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+from .. import results
+from . import Check
+
+
+IUSE_PREFIX = "ruby_targets_"
+
+
+class RubyCompatUpdate(results.VersionResult, results.Info):
+    """``USE_RUBY`` can be updated to support newer python version(s)."""
```
Comment (thesamesam) [inline src/pkgcheck/checks/ruby.py:15]:
```suggestion
    """``USE_RUBY`` can be updated to support newer Ruby version(s)."""
```

File: src/pkgcheck/checks/ruby.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        }
+
+    def feed(self, pkg):
+        if "ruby-ng" not in pkg.inherited:
+            return
+
+        deps = self.deps(pkg)
+
+        try:
+            # determine the latest supported python version
```
Comment (thesamesam) [inline src/pkgcheck/checks/ruby.py:74]:
```suggestion
            # determine the latest supported ruby version
```

File: src/pkgcheck/checks/ruby.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            latest_target = sorted(
+                (
+                    f"ruby{x.slot.replace('.', '')}"
+                    for x in deps
+                    if x.key == "dev-lang/ruby" and x.slot is not None
```
Comment (thesamesam) [inline src/pkgcheck/checks/ruby.py:79]: we should probably ignore virtual/rubygems and virtual/ruby-ssl too

Comment (arthurzam) [reply src/pkgcheck/checks/ruby.py:81]: done

**Review by thesamesam — APPROVED** [review]

**Pull request merged**